tools/pre-commit

```diff
@@ -1 +1,24 @@
-./tools/lint-all $(git diff --cached --name-only --diff-filter=ACM) || true
+#!/bin/sh
+
+# This hook runs the Zulip code linter ./tools/lint-all and returns true
+# regardless of linter results so that your commit may continue.
+
+# Messages from the linter will output to screen.
+#
+# If you are using the vagrant method for your Zulip dev setup, the linter will
+# be run through vagrant ssh.
+
+if `which vagrant > /dev/null` && vstatus="$(vagrant status)" && ! [[ $vstatus =~ 'not created' ]]
+then
+  vcmd='/srv/zulip/tools/lint-all $(cd /srv/zulip && git diff --cached --name-only --diff-filter=ACM) || true'
+  if ! [[ $vstatus =~ 'running' ]]
+  then
+    echo "Vagrant machine not running; lint-all skipped."
+  else
+    echo "Running lint-all using vagrant..."
+    vagrant ssh -c "$vcmd" 2> /dev/null
+  fi
+else
+  echo "Running lint-all..."
+  ./tools/lint-all $(git diff --cached --name-only --diff-filter=ACM) || true
+fi
```

Comment:
Good point about the timing. I think it makes sense to add a check for `.vagrantd` after `which vagrant` and before `vagrant status`. If `.vagrantd` isn't there, the vagrant command has never been run in the zulip directory and we can safely assume `lint-all` should be run directly, even if vagrant is installed. In this case `vagrant status` won't be run (I think, IIRC how BASH operators work) and won't take up those extra milliseconds.

Can you say more about checking for the vagrant user? The way I understand it, this tells you at most that you're running inside the dev virtual machine and lint-all should be run directly. But that's not the only circumstance for which lint-all should be run directly. If I've setup the dev environment manually on Ubuntu the username will be my username, which is usually not vagrant.